catalyst/dl/meters/ppv_tpr_f1_meter.py

```diff
@@ -0,0 +1,98 @@
+import numbers
+
+import numpy as np
+import torch
+
+from . import meter
+from collections import defaultdict
+
+
+class PrecisionRecallF1ScoreMeter(meter.Meter):
+    """
+    Keeps track of global true positives, false positives, and false negatives
+    for each epoch and calculates precision, recall, and F1-score based on
+    those metrics. Currently, this meter works for binary cases only, please
+    use multiple instances of this class for multi-label cases.
+    """
+    def __init__(self, threshold=0.5):
+        super(PrecisionRecallF1ScoreMeter, self).__init__()
+        self.threshold = threshold
+        self.reset()
+
+    def reset(self):
+        self.tp_fp_fn_counts = defaultdict(int)
+
+    def add(self, output, target):
+        if torch.is_tensor(output):
+            output = output.cpu().squeeze().numpy()
+        if torch.is_tensor(target):
+            target = target.cpu().squeeze().numpy()
+        elif isinstance(target, numbers.Number):
+            target = np.asarray([target])
+        assert np.ndim(output) == 1, \
+            "wrong output size (1D expected)"
+        assert np.ndim(target) == 1, \
+            "wrong target size (1D expected)"
+        assert output.shape[0] == target.shape[0], \
+            "number of outputs and targets does not match"
+        assert np.all(np.add(np.equal(target, 1), np.equal(target, 0))), \
+            "targets should be binary (0, 1)"
+
+        output = (output > self.threshold).astype(np.int32)
+
+        tp = np.sum(target * output)
+        fp = np.sum(output) - tp
+        fn = np.sum(target) - tp
+
+        self.tp_fp_fn_counts["tp"] += tp
+        self.tp_fp_fn_counts["fp"] += fp
+        self.tp_fp_fn_counts["fn"] += fn
+
+    def value(self):
+        precision_value = precision(self.tp_fp_fn_counts["tp"],
+                                    self.tp_fp_fn_counts["fp"])
+        recall_value = recall(self.tp_fp_fn_counts["tp"],
+                              self.tp_fp_fn_counts["fn"])
+        f1_value = f1score(precision_value, recall_value)
+        return (float(precision_value), float(recall_value), float(f1_value))
+
+
+def f1score(precision_value, recall_value, eps=1e-5):
+    """
+    Calculating F1-score from precision and recall to reduce computation
+    redundancy.
+    Args:
+        precision_value: precision (0-1)
+        recall_value: recall (0-1)
+    Returns:
+        F1 score (0-1)
+    """
+    numerator = 2 * (precision_value * recall_value)
+    denominator = precision_value + recall_value + eps
+    return numerator / denominator
+
+
+def precision(tp, fp, eps=1e-5):
+    """
+    Calculates precision (a.k.a. positive predictive value) for binary
+    classification and segmentation.
+    Args:
+        tp: number of true positives
+        fp: number of false positives
+    Returns:
+        precision value (0-1)
+    """
+    return tp / (tp + fp + eps)
```

Comment:
Could you please insert all used function in `PrecisionRecallF1ScoreMeter` on top of `PrecisionRecallF1ScoreMeter`? 
It would be nice to see everything before usage :)

Reply:
btw, could you please also add some mini test cases, to check the implementation correctness?
(we have some fun stories already)

Reply:
Just to clarify, do I make the test cases in dl/script and run it through an .sh file in catalyst/bin?

Reply:
you can make some test like [these one](https://github.com/catalyst-team/catalyst/tree/master/catalyst/dl/tests)